script.js

```diff
@@ -1,4 +1,117 @@
+var state = "INPUT PLAYER 1 NAME"
+var numOfPlayer1Wins = 0
+var numOfPlayer2Wins = 0
+var Player1UserName = ""
+var Player2UserName = ""
+var Player1Arr = []
+var Player2Arr = []
+
+
+
+var diceRoll = function () {
+  var randomInteger = Math.ceil(Math.random() * 6);
+  return randomInteger
+}
+
+var beat_that = function (input) {
+
+  if (state == "INPUT PLAYER 1 NAME") {
+    Player1UserName = input
+    state = "INPUT PLAYER 2 NAME"
+    return `Welcome, ${Player1UserName}. Please input Player 2's username to continue.`
+  }
+
+  else if (state == "INPUT PLAYER 2 NAME") {
+    Player2UserName = input
+    state = "PLAYER 1 DICE ROLL"
+    return `Welcome, ${Player2UserName}. Please click submit to roll the dice.`
+  }
+
+  else if (state == "PLAYER 1 DICE ROLL") {
+    while (Player1Arr.length < 2) {
+      Player1Arr.push(diceRoll())
+    }
+    state = "PLAYER 1 CHOOSE NUMBERS"
+    return `${Player1UserName} rolled <br>
+    ${Player1Arr[0]} for Dice 1 <br>
+    ${Player1Arr[1]} for Dice 2.<br> 
+      Choose the order of the dice.`
+  }
+
+  else if (state == "PLAYER 1 CHOOSE NUMBERS") {
+
+    if (input == 2) {
+      var tempNumber = Player1Arr[0]
+      Player1Arr[0] = Player1Arr[1]
+      Player1Arr[1] = tempNumber
+    }
+
+    var player1Numbers = Player1Arr[0].toString() + Player1Arr[1].toString()
+    state = "PLAYER 2 DICE ROLL"
+
+    return `${Player1UserName}, you chose Dice ${input} first. <br>
+          Your number is ${player1Numbers}. <br>
+          It is now Player 2's turn. Please click submit to roll the dice for player 2.`
+  }
+
+  else if (state == "PLAYER 2 DICE ROLL") {
+    while (Player2Arr.length < 2) {
+      Player2Arr.push(diceRoll())
+    }
+    state = "PLAYER 2 CHOOSE NUMBERS"
+    return `${Player2UserName} rolled<br>
+    ${Player2Arr[0]} for Dice 1 <br>
+    ${Player2Arr[1]} for Dice 2.<br> 
+      Choose the order of the dice.`
+  }
+
+  else if (state == "PLAYER 2 CHOOSE NUMBERS") {
+    if (input == 2) {
+      var tempNumber = Player2Arr[0]
+      Player2Arr[0] = Player2Arr[1]
+      Player2Arr[1] = tempNumber
+    }
+    var player2Numbers = Player2Arr[0].toString() + Player2Arr[1].toString()
+    state = "COMPARE WINNERS"
+
+    return `${Player2UserName}, you chose Dice ${input} first. <br>
+      Your number is ${player2Numbers}. <br>
+      Press submit to see who has the bigger number.`
+  }
+
+  else if (state == "COMPARE WINNERS") {
+    state = "PLAYER 1 DICE ROLL"
+    player1Numbers = Player1Arr[0].toString() + Player1Arr[1].toString()
+    player2Numbers = Player2Arr[0].toString() + Player2Arr[1].toString()
```

Comment:
A few things to note here: 
1. `player1Numbers` and `player2Numbers` were already defined and assigned in lines 49 and 74 respectively, so this is repeated code.
2. The reason why the earlier assignment doesn't carry through to this game mode is because they were defined as local variables (as opposed to global variables). As the `beat_that()` function is called in the `main` function, whose lifecycle ends on the click of the Submit button, the values stored in the variables defined in lines 49 and 74 'refresh' after the Submit button is clicked.
3. A better way of solving this might be to define `player1Numbers` and `player2Numbers` as a global variable, and do the assignment (without `var`) at lines 49 and 74. Then, you don't need to repeat the 2 lines of code here!